internal/executor/executor.go

```diff
@@ -215,7 +216,7 @@ func (executor *Executor) RunBuild(ctx context.Context) {
 	// Normal timeout-bounded context
 	timeout := time.Duration(response.TimeoutInSeconds) * time.Second
 
-	timeoutCtx, timeoutCtxCancel := context.WithTimeout(ctx, timeout)
+	timeoutCtx, timeoutCtxCancel := context.WithTimeoutCause(ctx, timeout, ErrTimedOut)
```

Comment:
Perhaps we can skip using the `WithTimeoutCause()` if we're never using the `context.Cause()`?

Reply:
Didn't read docs that deep and thought `ctx.Err()` will have it set. See [349540a](https://github.com/cirruslabs/cirrus-ci-agent/pull/363/commits/349540a7b10b8932cf0dfe96c5f30ade8f663fad)

Reply:
Sorry, but I'm still not sure why we need it if we already have `context.DeadlineExceeded` that serves the same purpose.